File: src/main/java/nl/hand/made/vending/machine/model/Role.java
package nl.hand.made.vending.machine.model;

public enum Role {

    ROLE_BUYER,
    ROLE_SELLER,
    ROLE_ADMIN;

    public static Role fromString(String stringValue) {
        for (Role role : Role.values()) {
            if (role.name().equalsIgnoreCase(stringValue)) {
                return role;
            }
        }

        return null;
    }

    @Override
    public String toString() {
        return this.name().toLowerCase();
    }
}

File: src/main/kotlin/nl/hand/made/vending/machine/model/Role.kt
package nl.hand.made.vending.machine.model

enum class Role {
    ROLE_BUYER, ROLE_SELLER;

    override fun toString() = name

    companion object {
        fun fromString(stringValue: String) =
            values().firstOrNull { role -> role.name.equals(stringValue, ignoreCase = true) }
    }
}
